lib/Pipeline/Runner.cpp

```diff
@@ -305,6 +305,12 @@ Error Runner::run(llvm::StringRef EndingStepName,
   if (ToExec.size() <= 1)
     return Error::success();
 
+  for (auto &StepGoalsPairs : llvm::drop_begin(ToExec)) {
+    auto &Step = *StepGoalsPairs.ToExecute;
+    if (auto Error = Step.precondition(getContext()); Error)
+      return Error;
```

Comment:
Shall we augment the error with step and pipe names?

Reply:
about that, i am usually kind of against augmenting errors, because it requires for the inner error to be converted to string. Do we have a policy on this?

Reply:
Eh we could wrap them instead of converting them to strings, but that gets Java-ish quite rapidly.
Maybe we can have the error with step and pipe fields that are filled in while returning?
It's rather important to know who generated the error.